src/components/Editor/GuestNameDialog.vue

```diff
@@ -60,19 +62,24 @@ export default {
 		},
 	},
 	beforeMount() {
-		this.guestName = this.syncService.guestName
+		this.guestName = this.session.guestName
 		this.updateBufferedGuestName()
 	},
+
 	methods: {
 		setGuestName() {
-			const previousGuestName = this.syncService.guestName
-			this.syncService
-				.updateSession(this.guestName)
+			if (!this.connection) {
+				showError(t('text', 'Not connected. Cannot update guest name.'))
```

Comment:
I guess you also want to throw here to prevent the subsequent code from being executed?